src/item/GlassBottle.php

```diff
@@ -23,6 +23,32 @@
 
 namespace pocketmine\item;
 
+use pocketmine\block\Block;
+use pocketmine\block\BlockLegacyIds;
+use pocketmine\event\player\PlayerDropItemEvent;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+
 class GlassBottle extends Item{
 
+	public function onInteractBlock(Player $player, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector) : ItemUseResult{
+		if($blockClicked->getId() === BlockLegacyIds::FLOWING_WATER){
+			$watterPotion = VanillaItems::WATER_POTION();
+			if($player->getInventory()->canAddItem($watterPotion)){
+				$player->getInventory()->addItem($watterPotion);
```

Comment:
This can be simplified by using the results of `addItem()` (it will return the items that didn't fit into the inventory).